alica_engine/src/engine/teammanager/TeamManager.cpp

```diff
@@ -232,18 +265,31 @@ AgentGrp TeamManager::updateAgents(bool changedSomeAgent)
     return deactivatedAgentIds;
 }
 
-void TeamManager::handleAgentQuery(const AgentQuery& pq) const
+void TeamManager::handleAgentQuery(const AgentQuery& aq) const
 {
-    //@759 check plan hash and sdk version before responding and log query here
-    if (!_useAutoDiscovery) {
+    if (!_useAutoDiscovery || aq.senderID == _localAgent->getId()) {
+        return;
+    }
+
+    // TODO: Add sdk compatibility check with comparing major version numbers
+    if (aq.senderSdk != _localAgent->getSdk() || aq.planHash != _localAgent->getPlanHash()) {
+        ALICA_DEBUG_MSG("tm: Version mismatch ignoring: " << aq.senderID);
         return;
     }
+
+    ALICA_DEBUG_MSG("tm: Responding to agent: " << aq.senderID);
     announcePresence();
 }
 
 void TeamManager::handleAgentAnnouncement(const AgentAnnouncement& aa)
 {
-    if (!_useAutoDiscovery) {
+    if (!_useAutoDiscovery || aa.senderID == _localAgent->getId()) {
+        return;
+    }
+
+    // TODO: Add sdk compatibility check with comparing major version numbers
+    if (aa.senderSdk != _localAgent->getSdk() || aa.planHash != _localAgent->getPlanHash()) {
```

Comment:
I would split these two checks into separate if blocks. Then you can print a message for both cases that is more helpful to the user - namely that either there was a version mismatch or a plan mismatch or both.

Reply:
same as above